pkg/encoding/v1/encoding.go

```diff
@@ -219,11 +222,81 @@ func (m *Mount) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	return nil
 }
 
+type Health struct {
+	ReadinessProbe interface{} `yaml:"readinessProbe,omitempty"`
+	Readiness      *api_v1.Probe
+
+	LivenessProbe interface{} `yaml:"livenessProbe,omitempty"`
+	Liveness      *api_v1.Probe
+}
+
+func interfaceToJSON(i interface{}) interface{} {
+	switch x := i.(type) {
+	case map[interface{}]interface{}:
+		m2 := map[string]interface{}{}
+		for k, v := range x {
+			m2[k.(string)] = interfaceToJSON(v)
+		}
+		return m2
+	case []interface{}:
+		for i, v := range x {
+			x[i] = interfaceToJSON(v)
+		}
```

Comment:
>The json package uses map[string]interface{} and []interface{} values to store arbitrary JSON objects and arrays; it will happily unmarshal any valid JSON blob into a plain interface{} value. 

So we don't need anything else more. Source: https://blog.golang.org/json-and-go